system/index.js

```diff
@@ -71,7 +71,10 @@ module.exports = async (message) => {
     }
   }));
 
-  middlewareResult = middlewareResult.concat(logger);
+  // ensure that a logger is added after middleware(s) are processed
+  if (middlewareFns.length > 0) {
```

Comment:
Do we need to add the logger here at all?  If there are no middlwares the logger is passed as the 2nd param...if there are middleware the signature should be 100% controlled by middleware results right?

Reply:
yeah, maybe we just have Jing add the logger to the result.  I like that idea.

Reply:
@jqian2017 how does this sound to you ^^ ?

Reply:
I thought @jqian2017 didn't even want to add `logger` as another argument. Isn't it part of the `context` being built and not an extra argument?